src/SassBinary.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+/*
+ * This file is part of the SymfonyCasts SassBundle package.
+ * Copyright (c) SymfonyCasts <https://symfonycasts.com/>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfonycasts\SassBundle;
+
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Component\Process\Process;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class SassBinary
+{
+    private const VERSION = '1.63.6';
+    private HttpClientInterface $httpClient;
+
+    public function __construct(
+        private string $binaryDownloadDir,
+        private ?string $binaryPath = null,
+        private ?SymfonyStyle $output = null,
+        HttpClientInterface $httpClient = null
+    ) {
+        $this->httpClient = $httpClient ?? HttpClient::create();
+    }
+
+    /**
+     * @param array<string> $args
+     */
+    public function createProcess(array $args): Process
+    {
+        if (null === $this->binaryPath) {
+            $binary = $this->binaryDownloadDir.'/dart-sass/sass';
+            if (!is_file($binary)) {
+                $this->downloadExecutable();
+            }
+        } else {
+            $binary = $this->binaryPath;
+        }
+
+        array_unshift($args, $binary);
+
+        return new Process($args);
+    }
+
+    public function downloadExecutable(): void
+    {
+        $url = sprintf('https://github.com/sass/dart-sass/releases/download/%s/%s.tar.gz', self::VERSION, $this->getBinaryName());
+
+        $this->output?->note('Downloading Sass binary from '.$url);
+
+        if (!is_dir($this->binaryDownloadDir)) {
+            mkdir($this->binaryDownloadDir, 0777, true);
+        }
+
+        $targetPath = $this->binaryDownloadDir.'/'.self::getBinaryName().'.tar.gz';
+        $progressBar = null;
+
+        $response = $this->httpClient->request('GET', $url, [
+            'on_progress' => function (int $dlNow, int $dlSize, array $info) use (&$progressBar): void {
+                if (0 === $dlSize) {
+                    return;
+                }
+
+                if (!$progressBar) {
+                    $progressBar = $this->output?->createProgressBar($dlSize);
+                }
+
+                $progressBar?->setProgress($dlNow);
+            },
+        ]);
+
+        $fileHandler = fopen($targetPath, 'w');
+        foreach ($this->httpClient->stream($response) as $chunk) {
+            fwrite($fileHandler, $chunk->getContent());
+        }
+
+        fclose($fileHandler);
+        $progressBar?->finish();
+        $this->output?->writeln('');
+        $archive = new \PharData($targetPath);
+        $archive->decompress();
+        $archive->extractTo($this->binaryDownloadDir);
+
+        unlink($targetPath);
+        unlink($this->binaryDownloadDir.'/'.self::getBinaryName().'.tar');
+
+        chmod($this->binaryPath, 0777);
```

Comment:
This fails because, by default, `$this->binaryPath` is null. This should actually reuse the logic from `createProcess()` where it uses `$this->binaryDownloadDir.'/dart-sass/sass';`.

Reply:
Also, we should move the binary into `var/sass` instead of `bin/...` as there is no reason for the user to wonder whether they should commit this file or not. This is how it's done in TailwindBundle - https://github.com/SymfonyCasts/tailwind-bundle/blob/main/config/services.php#L20C17-L20C22 - that's eventually passed to `new TailwindBinary()`: https://github.com/SymfonyCasts/tailwind-bundle/blob/main/src/TailwindBuilder.php#L101C42-L101C56

Reply:
I'm not sure how I feel about that. This pretty much breaks all existing conventions of having binaries in `bin`.
How about keeping in `bin` and adding a recipe to add a git ignore?

Reply:
The binary is a completely hidden thing - the user would never need to run it directly (and won't commit it). My thoughts: why put it in `bin/` for the user to see and worry about?